code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm

```diff
@@ -30,69 +53,74 @@
 /datum/action/xeno_action/toggle_bomb
 	name = "Toggle Bombard Type"
 	action_icon_state = "toggle_bomb0"
-	mechanics_text = "Switches Boiler Bombard type between Corrosive Acid and Neurotoxin."
+	mechanics_text = "Switches Boiler Bombard type between available glob types."
 	use_state_flags = XACT_USE_BUSY|XACT_USE_LYING
 	keybind_signal = COMSIG_XENOABILITY_TOGGLE_BOMB
+	alternate_keybind_signal = COMSIG_XENOABILITY_TOGGLE_BOMB_RADIAL
 
 /datum/action/xeno_action/toggle_bomb/can_use_action(silent = FALSE, override_flags)
 	. = ..()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
+	if(length(X.xeno_caste.spit_types) > 2)
+		return	//They might just be skipping past a invalid type
 	if((X.corrosive_ammo + X.neuro_ammo) >= X.xeno_caste.max_ammo)
-		if((X.ammo.type == /datum/ammo/xeno/boiler_gas/corrosive && X.neuro_ammo==0) || (X.ammo.type == /datum/ammo/xeno/boiler_gas && X.corrosive_ammo==0))
+		if((istype(X.ammo, /datum/ammo/xeno/boiler_gas/corrosive) && X.neuro_ammo==0) || (istype(X.ammo, /datum/ammo/xeno/boiler_gas) && X.corrosive_ammo==0))
```

Comment:
So this will probably not cause any bug in game, but (istype(X.ammo, /datum/ammo/xeno/boiler_gas) will be true even if X is corrosive

Reply:
true, returned to its previous state as it's only relevant for base boilers anyways.